app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -737,105 +737,6 @@ public TeiDownloadResult download(String teiUid, @Nullable String enrollmentUid,
-        }
-    }
-
-    private TeiDownloadResult breakTheGlass(String teiUid, String reason) {
-        if (downloadRepository != null) {
-            d2.trackedEntityModule().ownershipManager().blockingBreakGlass(teiUid, currentProgram, reason);
-            downloadRepository.overwrite(true).blockingDownload();
-            d2.fileResourceModule().blockingDownload();
-            downloadRepository = null;
-            return checkDownload(teiUid, null);
-        } else {
-            return new TeiDownloadResult.TeiNotDownloaded(teiUid);
-        }
-    }
-
-    private TeiDownloadResult checkDownload(String teiUid, @Nullable String enrollmentUid) {
-        if (teiHasBeenDownloaded(teiUid)) {
-            if (hasEnrollmentInCurrentProgram(teiUid)) {
-                String programEnrollment;
-                if (enrollmentUid != null) {
-                    programEnrollment = enrollmentUid;
-                } else {
-                    programEnrollment = getEnrollmentInProgram(teiUid);
-                }
-                return new TeiDownloadResult.DownloadedResult(teiUid, currentProgram, programEnrollment);
-            } else if (canEnrollInCurrentProgram()) {
-                return new TeiDownloadResult.TeiToEnroll(teiUid);
-            } else {
-                return new TeiDownloadResult.DownloadedResult(teiUid, currentProgram, null);
-            }
-        } else {
-            return new TeiDownloadResult.TeiNotDownloaded(teiUid);
-        }
-    }
-
-    private boolean teiHasBeenDownloaded(String teiUid) {
-        return d2.trackedEntityModule().trackedEntityInstances().uid(teiUid).blockingExists();
-    }
-
-    private boolean hasEnrollmentInCurrentProgram(String teiUid) {
-        if (currentProgram != null) {
-            return !d2.enrollmentModule().enrollments()
-                    .byTrackedEntityInstance().eq(teiUid)
-                    .byProgram().eq(currentProgram)
-                    .blockingIsEmpty();
-        } else {
-            return false;
-        }
-    }
-
-    private String getEnrollmentInProgram(String teiUid) {
-        return d2.enrollmentModule().enrollments()
-                .byTrackedEntityInstance().eq(teiUid)
-                .byProgram().eq(currentProgram)
-                .one()
-                .blockingGet()
-                .uid();
-    }
-
-    private boolean canEnrollInCurrentProgram() {
-        if (currentProgram == null) return false;
-        Program selectedProgram = d2.programModule().programs().uid(currentProgram).blockingGet();
-        boolean programAccess = selectedProgram.access().data().write() != null && selectedProgram.access().data().write();
-        boolean teTypeAccess = d2.trackedEntityModule().trackedEntityTypes().uid(
-                selectedProgram.trackedEntityType().uid()
-        ).blockingGet().access().data().write();
-        return programAccess && teTypeAccess;
-    }
-
```

Comment:
Were we not using all of this?

Reply:
It was all moved to the TeiDownloader class. We forgot to remove the unused code